src/globus_cli/commands/gcs/endpoint/set_subscription_id.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import typing as t
+import uuid
+
+import click
+
+from globus_cli.constants import EXPLICIT_NULL, ExplicitNullType
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.termio import TextMode, display
+
+
+class GCSSubscriptionIdType(click.ParamType):
+    def get_type_annotation(self, _: click.Parameter) -> type:
+        return t.cast(type, uuid.UUID | t.Literal["DEFAULT"] | ExplicitNullType)
+
+    def convert(
+        self, value: str, param: click.Parameter | None, ctx: click.Context | None
+    ) -> t.Any:
+        if value is None or (ctx and ctx.resilient_parsing):
+            return None
+        if value.lower() == "null":
+            return EXPLICIT_NULL
+        elif value.lower() == "default":
+            return "DEFAULT"
+        try:
+            uuid.UUID(value)
+            return value
+        except ValueError:
+            msg = (
+                f"{value} is invalid. Expected either a UUID or the special "
+                'values "DEFAULT" or "null"'
+            )
+            self.fail(msg, param, ctx)
+
+
+@command("set-subscription-id", short_help="Update an endpoint's subscription")
```

Comment:
Short help differs from first line of docstring:

```suggestion
@command("set-subscription-id", short_help="Update an endpoint's subscription ID")
```

Reply:
Syncing in the opposite direction (updating the first line of the docstring.